### ISSUE TITLE
Upgrading from <0.33.1 to >=0.33.1 requires state move for bucket

### ISSUE THREAD

**Issue by ScottMillard** [cloudposse/terraform-aws-tfstate-backend#110]
## Describe the Bug
When upgrading from before version `0.33.1` to any version after `0.33.1`:
count was [added](https://github.com/cloudposse/terraform-aws-tfstate-backend/pull/92/files#diff-dc46acf24afd63ef8c556b77c126ccc6e578bc87e3aa09a931f33d9bf2532fbbR108) to the `default` bucket so a plan sees that the bucket does not exist and wants to remove it from state. 

No where in the release notes or README does it mention needing to do a state move.  Once you move the state to add an index a plan will show no changes needed:
`terraform state mv module.tfstate-backend.aws_s3_bucket.default module.tfstate-backend.aws_s3_bucket.default[0]`

## Expected Behavior
A note in the 0.33.1 release about needing to perform the state move.

## Steps to Reproduce
1. Init a terraform project with `terraform-aws-tfstate-backend` version `0.33.0`
2. Run terraform apply
2. Change the `terraform-aws-tfstate-backend` to version `0.33.1`
3. Run terraform plan

## Screenshots
N/A

## Environment (please complete the following information):
```
Terraform v1.0.5
on darwin_amd64
+ provider registry.terraform.io/hashicorp/aws v3.56.0
```

**Comment by nitrocode** [comment]
I thought newer versions of terraform correct for `count` logic added or subtracted from components as long as the `count` is either `1` or `0`. Perhaps it is for a terraform version greater than `1.0.5`?

**Comment by Nuru** [comment]
Closing as fixed in Terraform v1.1. See [note in Terraform documentation](https://developer.hashicorp.com/terraform/language/modules/develop/refactoring#enabling-count-or-for_each-for-a-resource):

> When you add `count` to an existing resource that didn't use it, Terraform automatically proposes to move the original object to instance zero, unless you write an moved block explicitly mentioning that resource.
